dist/src/main/java/io/camunda/identity/webapp/controllers/IdentityClientConfigController.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.identity.webapp.controllers;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.camunda.security.configuration.SecurityConfiguration;
+import io.camunda.security.entity.AuthenticationMethod;
+import io.swagger.v3.oas.annotations.Hidden;
+import java.util.Map;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.ResponseBody;
+
+@Controller
+public class IdentityClientConfigController {
+
+  private static final Logger LOG = LoggerFactory.getLogger(IdentityClientConfigController.class);
+
+  private static final String VITE_IS_OIDC = "VITE_IS_OIDC";
+  private static final String VITE_INTERNAL_GROUPS_ENABLED = "VITE_INTERNAL_GROUPS_ENABLED";
+
+  private String clientConfigAsJS;
+
+  public IdentityClientConfigController(final SecurityConfiguration securityConfiguration) {
+    try {
+      clientConfigAsJS =
+          String.format(
+              "window.clientConfig = %s;",
+              new ObjectMapper()
+                  .writeValueAsString(
+                      Map.of(
+                          VITE_IS_OIDC,
+                          String.valueOf(
+                              AuthenticationMethod.OIDC.equals(
+                                  securityConfiguration.getAuthentication().getMethod())),
+                          VITE_INTERNAL_GROUPS_ENABLED,
+                          String.valueOf(
+                              securityConfiguration.getAuthentication().getOidc() == null
+                                  || securityConfiguration
+                                          .getAuthentication()
+                                          .getOidc()
+                                          .getGroupsClaim()
+                                      == null))));
```

Comment:
This is a veeeeery long statement that's not particularly easy to read. Could you please pull it a bit apart? E.g. separately instantiating the object mapper, extracting the config values into local variables?

Sorry for not mentioning this on my first review already.

Reply:
Sure, done